DataFormats/Scouting/interface/ScoutingMuon.h

```diff
@@ -12,164 +13,290 @@ class ScoutingMuon {
+  unsigned int recoMuonRPClayerMask() const { return recoMuonRPClayerMask_; }
   int nValidPixelHits() const { return nValidPixelHits_; }
-  int nMatchedStations() const { return nMatchedStations_; }
-  int nTrackerLayersWithMeasurement() const { return nTrackerLayersWithMeasurement_; }
-  int type() const { return type_; }
-  bool isGlobalMuon() const { return type_ & 1 << 1; }
-  bool isTrackerMuon() const { return type_ & 1 << 2; }
   int nValidStripHits() const { return nValidStripHits_; }
+  int nPixelLayersWithMeasurement() const { return nPixelLayersWithMeasurement_; }
+  int nTrackerLayersWithMeasurement() const { return nTrackerLayersWithMeasurement_; }
+  float trk_chi2() const { return trk_chi2_; }
+  float trk_ndof() const { return trk_ndof_; }
+  float trk_dxy() const { return trk_dxy_; }
+  float trk_dz() const { return trk_dz_; }
   float trk_qoverp() const { return trk_qoverp_; }
   float trk_lambda() const { return trk_lambda_; }
   float trk_pt() const { return trk_pt_; }
   float trk_phi() const { return trk_phi_; }
   float trk_eta() const { return trk_eta_; }
-  float dxyError() const { return dxyError_; }
-  float dzError() const { return dzError_; }
+  float trk_dxyError() const { return trk_dxyError_; }
+  float trk_dzError() const { return trk_dzError_; }
   float trk_qoverpError() const { return trk_qoverpError_; }
   float trk_lambdaError() const { return trk_lambdaError_; }
   float trk_phiError() const { return trk_phiError_; }
   float trk_dsz() const { return trk_dsz_; }
   float trk_dszError() const { return trk_dszError_; }
+  //add off-diagonal covariance matrix parameter, the above "Error" variables correspond to the diagonal, enum for Cov matrix (qoverp, lambda, phi, dxy, dsz), see https://github.com/cms-sw/cmssw/blob/CMSSW_11_2_X/DataFormats/TrackReco/src/TrackBase.cc for details
+  float trk_qoverp_lambda_cov() const { return trk_qoverp_lambda_cov_; }
+  float trk_qoverp_phi_cov() const { return trk_qoverp_phi_cov_; }
+  float trk_qoverp_dxy_cov() const { return trk_qoverp_dxy_cov_; }
+  float trk_qoverp_dsz_cov() const { return trk_qoverp_dsz_cov_; }
+  float trk_lambda_phi_cov() const { return trk_lambda_phi_cov_; }
+  float trk_lambda_dxy_cov() const { return trk_lambda_dxy_cov_; }
+  float trk_lambda_dsz_cov() const { return trk_lambda_dsz_cov_; }
+  float trk_phi_dxy_cov() const { return trk_phi_dxy_cov_; }
+  float trk_phi_dsz_cov() const { return trk_phi_dsz_cov_; }
+  float trk_dxy_dsz_cov() const { return trk_dxy_dsz_cov_; }
+  float trk_vx() const { return trk_vx_; }
+  float trk_vy() const { return trk_vy_; }
+  float trk_vz() const { return trk_vz_; }
+  reco::HitPattern trk_hitPattern() const { return trk_hitPattern_; }
   std::vector<int> vtxIndx() const { return vtxIndx_; }
 
 private:
   float pt_;
   float eta_;
   float phi_;
   float m_;
+  unsigned int type_;
+  int charge_;
+  float normalizedChi2_;
   float ecalIso_;
   float hcalIso_;
   float trackIso_;
-  float chi2_;
```

Comment:
how is reading old raw scouting data affected by these changes? Are I/o rules not needed?

Reply:
also adding @mmasciov 

Renaming and (possibly) order of variables could break backward release compatibility. I think we have 3 options:
1.) break compatibility
2.) reverse the renaming of the variables affected and "append" new ones in this PR
3.) create new DataFormats for Run-3 development, that will be changed and developed until 2022 for data-taking. This will likely mean that we will also have to create new Scouting Producers.

Best will be 3.) or 2.). I am not sure how long 3.) will take us to implement